libs/labelbox/src/labelbox/schema/user_group.py

```diff
@@ -0,0 +1,523 @@
+
+        Returns:
+            Group: The newly created group. (self)
+
+        Raises:
+            ResourceCreationError: If the group already exists or if the creation fails.
+            ValueError: If the group name is not provided.
+        """
+        if self.id:
+            raise ResourceCreationError("Group already exists")
+        if not self.name:
+            raise ValueError("Group name is required")
+        query = """
+        mutation CreateUserGroupPyApi($name: String!, $color: String!, $projectIds: [String!]!, $userIds: [String!]!) {
+            createUserGroup(
+                data: {
+                    name: $name,
+                    color: $color,
+                    projectIds: $projectIds,
+                    userIds: $userIds
+                }
+            ) {
+                group {
+                    id
+                    name
+                    color
+                    projects {
+                        nodes {
+                            id
+                            name
+                        }
+                    }
+                    members {
+                        nodes {
+                            id
+                            email
+                        }
+                    }
+                }
+            }
+        }
+        """
+        params = {
+            "name":
+                self.name,
+            "color":
+                self.color.value,
+            "projectIds": [
+                project.id if hasattr(project, 'id') else project.uid
+                for project in self.projects
+            ],
+            "userIds": [
+                user.id if hasattr(user, 'id') else user.uid
+                for user in self.users
+            ]
+        }
+        result = self.client.execute(query, params)
+        if not result:
+            raise ResourceCreationError("Failed to create group")
+        result = result["createUserGroup"]["group"]
+        self.id = result["id"]
+        return self
+
+    def delete(self) -> bool:
+        """
+        Deletes the user group from Labelbox.
+
+        This method sends a mutation request to the Labelbox API to delete the user group
+        with the specified ID. If the deletion is successful, it returns True. Otherwise,
+        it raises an UnprocessableEntityError and returns False.
+
+        Returns:
+            bool: True if the user group was successfully deleted, False otherwise.
+        
+        Raises:
+            UnprocessableEntityError: If the deletion of the user group fails.
+        """
+        query = """
+        mutation DeleteUserGroupPyApi($id: ID!) {
+            deleteUserGroup(where: {id: $id}) {
+                success
+            }
+        }
+        """
+        params = {"id": self.id}
+        result = self.client.execute(query, params)
+        if not result:
+            raise UnprocessableEntityError("Failed to delete user group")
```

Comment:
should we add a more detailed error message from the server here too?

Reply:
The server will raise the error specifically (e.g. client.execute itself will throw the error), this is just a general guard aginst execute returning none in any case